resources/app/src/components/groups/MemberForm.vue

```diff
@@ -257,13 +223,15 @@ export default {
         }
     },
     watch: {
-        group () {
+        group() {
             this.syncMember();
         },
         'newMember.roles': {
             deep: true,
-            handler (to, from) {
-                if (to.map(r => r.name).includes('coordinator') && !from.map(r => r.name).includes('coordinator')) {
+            handler(to, from) {
+                const hasCoordinator = to.map(r => r.name).includes('coordinator');
+                const hasGrantLiaison = to.map(r => r.name).includes('grant-liaison');
+                if ((hasCoordinator || hasGrantLiaison) && !from.some(r => r.name === 'coordinator' || r.name === 'grant-liaison')) {
```

Comment:
This looks odd to me. Couldn't you be consistent between the checking of the "to" and "from" cases? Something like:
```
if (to.some(r => r.name === 'coordinator' || r.name === 'grant-liaison') &&
    !from.some(r => r.name === 'coordinator' || r.name === 'grant-liaison')) {
```

That does get to be somewhat difficult to read, so could even consider having a "helper function" for "containsNamedRole" to be a bit more "self-documenting", like:
```
function containsNamedRole(roles, needle) {
  return roles.map(r => r.name).includes(needle);
}
```

Reply:
The code is logical to me.  It sets variables for the values in question.

Reply:
I guess what I am trying to get across is that the logic of this is "set is_contact to true if the 'to' roles include either 'coordinator' or 'grant-liaison' but the 'from' roles didn't contain either of these roles". So the overall structure is "'to' roles meet this criteria, but 'from' roles do not".

But the way you have it written, the evaluation of "contains 'coordinator' or 'grant-liaison'" is structured differently to the 'to' side and the negated 'from' side. In reading it, it was less clear what the logic was and I had to think more about it, which means that any future developer would also have that additional cognitive load. My comment is about readability and maintainability.

That being said, the logic here might also be overly complicated in the way it already was when it was just for coordinator, and maybe we should re-assess with with Courtney whether we need to be this specific. Maybe we'd be better off just setting "is_contact" when the 'to' roles contain 'coordinator' or 'grant-liaison' and not worrying about what the 'from' roles were.